Microsoft.Toolkit.Uwp.SampleApp/Models/Sample.cs

```diff
@@ -215,14 +223,88 @@ public bool IsSupported
                     {
                         var result = await localDocsStream.ReadTextAsync();
                         _cachedDocumentation = ProcessDocs(result);
+                        _cachedPath = localPath;
                     }
                 }
                 catch (Exception)
                 {
                 }
             }
 
-            return _cachedDocumentation;
+            return (_cachedDocumentation, _cachedPath);
+        }
+
+        /// <summary>
+        /// Gets the image data from a Uri, with Caching.
+        /// </summary>
+        /// <param name="uri">Image Uri</param>
+        /// <returns>Image Stream</returns>
+        public async Task<IRandomAccessStream> GetImageStream(Uri uri)
+        {
+            async Task<Stream> CopyStream(HttpContent source)
+            {
+                var stream = new MemoryStream();
+                await source.CopyToAsync(stream);
+                stream.Seek(0, SeekOrigin.Begin);
+                return stream;
+            }
+
+            IRandomAccessStream imageStream = null;
+            var localpath = $"{uri.Host}/{uri.LocalPath}";
+
+            // Cache only in Release
+#if !DEBUG
+                try
+                {
+                    imageStream = await StreamHelper.GetLocalCacheFileStreamAsync(localpath, Windows.Storage.FileAccessMode.Read);
+                }
+                catch
+                {
+                }
+#endif
+
+            if (imageStream == null)
+            {
+                try
+                {
+                    using (var response = await client.GetAsync(uri))
+                    {
+                        if (response.IsSuccessStatusCode)
+                        {
+                            var imageCopy = await CopyStream(response.Content);
+                            imageStream = imageCopy.AsRandomAccessStream();
+
+                            // Cache only in Release
+#if !DEBUG
+                                // Takes a second copy of the image stream, so that is can save the image data to cache.
+                                SaveImageToCache(localpath, await CopyStream(response.Content));
+#endif
+                        }
+                    }
+                }
+                catch
+                {
+                }
+            }
+
+            return imageStream;
+        }
+
+        private async void SaveImageToCache(string localpath, Stream imageStream)
```

Comment:
`async void` methods should be a**void**ed. I understand we want to get the image back to the user as fast as possible, but this only happens once. Every time after it should grab from the local file system. 
Either
1) Make this a Task returning method and await it
2) Change `CopyToAsync` to `CopyTo`. I rarely notice a performance hit when coping from one stream to the next.